### PR TITLE
Fix struct definitions

### PR DESCRIPTION
**Pull request by saulrh**
The big change here is that I've added typdefs to all of the struct definitions. This has two effects.

First, there were a few struct definitions of the form `struct type { } type_t`, missing the typedef that this pattern typically begins with. This code, instead of creating a type type_t that you can use instead of `struct type`, creates a _variable_ named type_t of type struct type. This is annoying and clutters up namespaces, but the bigger problem is that this typo will completely break things if you try to use hubo-ach from a c++ program. When the variable type_t is defined, C++ will attempt to call its undefined constructor and break; adding the typedef fixes this behaviour.

Second, the codebase was littered with variable and function definitions which used the full `struct type var` thing to declare variables of struct types. Adding the typedefs means that you can use `type_t var` instead of `struct type var`, which makes those long function definitions a lot easier to understand.

TL;DR: you can use "type_t" instead of "struct type", and hubo-ach now works with c++ projects.

### CODE REVIEW

**Comment by a-price** [comment]
Saul, thanks for fixing this. It causes some rather confusing c++ linker errors about multiple definitions of the various structs.

**Comment by thedancomplex** [comment]
I was not having issues with hubo-console.cpp  which is a C++ project.  Why does it work for that and not for you other programs?

I ask because this is a major change which will cause people to modify their own code that already exists (which is not desirable)

**Comment by ndantam** [comment]
On 02/23/2013 03:24 PM, thedancomplex wrote:

> I was not having issues with hubo-console.cpp  which is a C++
> project.  Why does it work for that and not for you other
> programs?

Probably hubo-console was only a single source file?  The link
errors happen when linking together multiple object files (*.o) into
an executable.

> I ask because this is a major change which will cause people to
> modify their own code that already exists (which is not desirable)

The missing typedefs were definately errors and need to be fixed.
Using "foo_bar_t" instead of "struct foo_bar" doesn't really matter.
People can continue using either in the code.

-ntd

**Comment by thedancomplex** [comment]
Wonderful, this will be in the next release!  Thank you!

**Comment by saulrh** [comment]
Added a few more commits to pick up some more struct statements. I completely missed four types of structs, and then I learned that sed only replaces one match per line unless instructed. The only structs left now should be timespecs, can_frames, and one-offs from various other places.

**Comment by thedancomplex** [comment]
If you are going to move the struct hubo_state to hubo_state_t then please move struct hubo_ref to hubo_ref_t and other structs to use the same convention.  If it is only done for some and not all I do not see the point

**Comment by thedancomplex** [comment]
I changed all struct hubo_xxx to hubo_xxx_t in the feature/addCurrentSens branch

**Comment by saulrh** [comment]
I did move struct hubo_ref to struct hubo_ref_t; that was commit b89c1ee. The extra commits I added on 2013-02-26 were to do that since I'd missed them when I was fixing the other typedef problems on 2013-12-22. Sorry if you'd pulled those changes and the commits on the 26th were duplicated; I wouldn't have been able to see it if you pulled into a branch instead of to main.
